### PR TITLE
Change UUID module to address WS-2018-0594

### PR DESCRIPTION
**Pull request by jewzaam**
### Which issue this PR addresses:

Fixes https://www.whitesourcesoftware.com/vulnerability-database/WS-2018-0594

### What this PR does / why we need it:

Move from github.com/satori/go.uuid to github.com/gofrs/uuid for uuid generation.  Requires additional tweaks.
Dependency azure-sdk-for-go needed updated, it is moved to latest release.
Signature of NewV4() changed to return multiple values.  Created facade in pkg/util/uuid to allow us to continue using in-line with single argument returned.

### Test plan for issue:

Existing unit and integration tests.

### Is there any documentation that needs to be updated for this PR?

No, internal dependency update.

### CODE REVIEW

File: pkg/util/uuid/uuid.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import (
+	"github.com/gofrs/uuid"
+)
+
+// This is a facade to aid in conversion from github.com/satori/go.uuid to github.com/gofrs/uuid
+// Changing due to https://www.whitesourcesoftware.com/vulnerability-database/WS-2018-0594
+
+// helper so we can in-line uuid generation with gofrs, drops the error that would be returned
+func NewV4() uuid.UUID {
+	u, _ := uuid.NewV4()
```
Comment (jim-minter) [inline pkg/util/uuid/uuid.go:12]: must read the error and panic if it is set; not to do so here could be a security risk

File: pkg/util/uuid/uuid.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+// simple passthrough
+func Must(u uuid.UUID, err error) uuid.UUID {
+	return uuid.Must(u, err)
+}
+
+// simple passthroughs
+func FromString(input string) (uuid.UUID, error) {
+	return uuid.FromString(input)
+}
```
Comment (jim-minter) [inline pkg/util/uuid/uuid.go:24]: please set your editor to add trailing newlines to files

File: hack/validate-imports/validate-imports.go [hunk 3]
```diff
@@ -227,7 +227,7 @@ nextImport:
 			errs = append(errs, fmt.Errorf("%s is imported; use github.com/ghodss/yaml", value))
 			continue nextImport
 		case "github.com/google/uuid":
```
Comment (jim-minter) [inline hack/validate-imports/validate-imports.go:229]:
```suggestion
		case "github.com/google/uuid", "github.com/satori/go.uuid":
```

File: hack/validate-imports/validate-imports.go [hunk 4]
```diff
@@ -154,7 +154,7 @@ func acceptableNames(path string) []string {
 		return []string{"icazure"}
 	case "github.com/openshift/installer/pkg/types/azure":
 		return []string{"azuretypes"}
-	case "github.com/satori/go.uuid":
+	case "github.com/gofrs/uuid":
 		return []string{"uuid"}
```
Comment (jim-minter) [inline hack/validate-imports/validate-imports.go:158]: remove these lines

File: pkg/util/uuid/uuid.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package uuid
+
+import (
+	"github.com/gofrs/uuid"
+)
+
+// This is a facade to aid in conversion from github.com/satori/go.uuid to github.com/gofrs/uuid
```
Comment (jim-minter) [inline pkg/util/uuid/uuid.go:7]: +1 on using a facade

**Review by jim-minter — CHANGES_REQUESTED** [review]

File: pkg/util/uuid/uuid.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package uuid
+
```
Comment (mjudeikis) [inline pkg/util/uuid/uuid.go:2]: You are missing Microsoft license here. Run all tests locally for it to be generated or just add it from other files.

**Comment by jewzaam** [comment]
/hold

cloud-credential-operator dependency relies on github.com/satori/go.uuid so it's still being referenced in go.sum

**Comment by jim-minter** [comment]
merging; given that actually uuid.Must() exists let's follow up and remove pkg/util/uuid and standardise further.

**Pull request merged**